lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java

```diff
@@ -158,14 +173,52 @@ WordCase caseOf(char[] word, int length) {
     return null;
   }
 
+  List<char[]> sharpSVariations(char[] word, int length) {
+    if (!dictionary.checkSharpS) return Collections.emptyList();
+
+    Stream<String> result =
```

Comment:
Do you think it makes sense to use language trickery (anonymous subclass of Object) just to hide those two methods? I see it's a recursive call to replaceSS but I'd just move it to static methods at the top class level and let them be used directly from there?

Reply:
I also wonder if all that stream trickery is worth it in favor of a simple collector list pushed down recursively... Streams are nice, but a list requires less thinking (to me).

Reply:
The anonymous class serves two purposes: avoiding auxiliary methods polluting class namespace and avoiding passing `char[] word, int length` in them. If you think the verbosity isn't worth that, I can make the methods static.

Streams actually seem more natural to me for such a recursive enumeration. I can do the same with lists, but that will probably be a bit more verbose (and probably allocate a bit more, although that shouldn't matter much here). Should I?

Reply:
I understand the motives. But I tend to believe 25% of Java developers who see this construct will scratch their heads thinking what it does. :)

Once you start using streams, memory allocation is no longer a concern... (static methods and argument passing would definitely not allocate more). 

It's fine, leave it.

Reply:
They might. But that might be educating and doesn't seem too elaborate :)